### PR TITLE
 Add "Tenant" column in "vmdkops_admin ls" output and fix "vmdkops_admin set" command

### LINKED ISSUE #964: [Tenant] Add tenant column to admin ls commands
There should be way to extract volume<->tenant association information. It would be more informative for user to know a particular volume is part of which tenant. 

Proposed solution as per offline discussion with @msterin: ```vmdkops_admin ls``` should have a column displaying information about the volume resided into tenant. 

//CC @msterin @pdhamdhere

### LINKED ISSUE #998: Unable to modify access attribute for volume from admin-cli
Test steps:
1. Created a volume with access set to read-only
2. Tried changing access to read-write from admin cli but failed - Failed to update access=read-write for Vol100MB@sharedVmfs-0.

Logs are attached at the end.

----------------------
Respective commands and their outputs are as follows:

Created a volume with access mode set to "read-only"

```
root@sc-rdops-vm02-dhcp-52-237:/# docker volume create --driver=vsphere --name=Vol100MB -o size=100mb -o attach-as=persistent -o access=read-only -o diskformat=thin -o fstype=ext4
Vol100MB
root@sc-rdops-vm02-dhcp-52-237:/# docker volume ls
DRIVER              VOLUME NAME
vsphere             Vol100MB@sharedVmfs-0
root@sc-rdops-vm02-dhcp-52-237:/# docker volume inspect Vol100MB@sharedVmfs-0
[
    {
        "Name": "Vol100MB@sharedVmfs-0",
        "Driver": "vsphere",
        "Mountpoint": "/mnt/vmdk/Vol100MB@sharedVmfs-0",
        "Status": {
            "access": "read-only",  <<<<<<<<<<<<<<<<<<<<<<<<<
            "attach-as": "persistent",
            "capacity": {
                "allocated": "13MB",
                "size": "100MB"
            },
            "clone-from": "None",
            "created": "Mon Mar  6 21:51:21 2017",
            "created by VM": "ubuntu-VM1.3",
            "datastore": "sharedVmfs-0",
            "diskformat": "thin",
            "fstype": "ext4",
            "status": "detached"
        },
        "Labels": {},
        "Scope": "global"
    }
]

```

Admin cli output listing the newly created volume:

```
[root@sc-rdops-vm01-dhcp-2-37:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Volume    Datastore     Created By VM  Created                   Attached To VM (name/uuid)  Policy  Capacity  Used  Disk Format  Filesystem Type  Access     Attach As
--------  ------------  -------------  ------------------------  --------------------------  ------  --------  ----  -----------  ---------------  ---------  ----------
Vol100MB  sharedVmfs-0  ubuntu-VM1.3   Mon Mar  6 21:51:21 2017  ubuntu-VM0.3                N/A     100MB     13MB  thin         ext4             read-only  persistent

```

Tried changing the access attribute for volume to "read-write" but unsuccesful.

```
[root@sc-rdops-vm01-dhcp-2-37:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py set --volume=Vol100MB@sharedVmfs-0 --options="access=read-write"
Failed to update access=read-write for Vol100MB@sharedVmfs-0.
```

vmdk_ops.log:
```
03/06/17 22:03:38 77393 [MainThread] [WARNING] Volume Vol100MB not found.
03/06/17 22:05:46 77424 [MainThread] [INFO   ] Connecting to the local Service Instance as 'dcui'
03/06/17 22:06:11 77431 [MainThread] [WARNING] Volume Vol100MB not found.
03/06/17 22:06:30 77436 [MainThread] [WARNING] Invalid datastore 'None'.

03/06/17 22:10:39 77459 [MainThread] [WARNING] Volume Vol100MB not found.
```

----------
Logs:
[vmdk_ops.txt](https://github.com/vmware/docker-volume-vsphere/files/822716/vmdk_ops.txt)
[docker-volume-vsphere.txt](https://github.com/vmware/docker-volume-vsphere/files/822717/docker-volume-vsphere.txt)

### PR DESCRIPTION
**Pull request by lipingxue**
This PR includes:
1. Add a new column "Tenant" in `vmdkops_admin ls` command (fixes #964)
2. Change to `vmdkops_admin set` command which take one more argument "--tenant". (fixes #998)
3. admin-cli.md is changed accordingly.

### CODE REVIEW

**Comment by lipingxue** [comment]
Test for add column "Tenant" for "vmdkops_admin ls" output.

I have done the following test manually:
Configuration:
_DEFAULT tenant and a named tenant "tenant1"
create two volumes with the same name "vol1", which belongs to two different tenant "_DEFAULT" and "tenant1" accordingly

Step1:
```
[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Tenant    Volume    Datastore   Created By VM  Created                   Attached To VM (name/uuid)  Policy  Capacity  Used  Disk Format  Filesystem Type  Access      Attach As              
--------  --------  ----------  -------------  ------------------------  --------------------------  ------  --------  ----  -----------  ---------------  ----------  ---------------------- 
_DEFAULT  vol1      datastore1  photon-6       Sat Sep 10 01:11:07 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  independent_persistent 
_DEFAULT  testvol   datastore1  photon-6       Sat Sep 10 04:21:02 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  independent_persistent 
_DEFAULT  Vol100MB  datastore1  photon-6       Sat Sep 10 08:36:29 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  persistent             
tenant1   vol1      datastore1  photon-6       Sat Sep 10 09:36:51 2016  detached                    N/A     500MB     23MB  thin         ext4             read-write  independent_persistent 
```

Step2: remove "tenant1" without removing volumes, "vol1" which belongs to "tenant1" still show up in the output, but column "Tenant" for that volume shows as "N/A" since "tenant1" has been removed.
```
[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py tenant ls
Uuid                                  Name      Description               Default_datastore  VM_list          
------------------------------------  --------  ------------------------  -----------------  ---------------- 
11111111-1111-1111-1111-111111111111  _DEFAULT  This is a default tenant                                      
d91ad806-a9e6-447f-a464-fbd5314db85f  tenant1                                                photon7,photon-6 
 
[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py tenant rm --name=tenant1
tenant rm succeeded

[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Tenant    Volume    Datastore   Created By VM  Created                   Attached To VM (name/uuid)  Policy  Capacity  Used  Disk Format  Filesystem Type  Access      Attach As              
--------  --------  ----------  -------------  ------------------------  --------------------------  ------  --------  ----  -----------  ---------------  ----------  ---------------------- 
_DEFAULT  vol1      datastore1  photon-6       Sat Sep 10 01:11:07 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  independent_persistent 
_DEFAULT  testvol   datastore1  photon-6       Sat Sep 10 04:21:02 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  independent_persistent 
_DEFAULT  Vol100MB  datastore1  photon-6       Sat Sep 10 08:36:29 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  persistent             
N/A       vol1      datastore1  photon-6       Sat Sep 10 09:36:51 2016  detached                    N/A     500MB     23MB  thin         ext4             read-write  independent_persistent 
```

**Comment by lipingxue** [comment]
Test for add argument "--tenant" for  `vmdkops_admin set` command
Step1: run `vmdkops_admin ls` command
```
[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Tenant    Volume    Datastore   Created By VM  Created                   Attached To VM (name/uuid)  Policy  Capacity  Used  Disk Format  Filesystem Type  Access      Attach As              
--------  --------  ----------  -------------  ------------------------  --------------------------  ------  --------  ----  -----------  ---------------  ----------  ---------------------- 
_DEFAULT  vol1      datastore1  photon-6       Sat Sep 10 01:11:07 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  independent_persistent 
_DEFAULT  testvol   datastore1  photon-6       Sat Sep 10 04:21:02 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  independent_persistent 
_DEFAULT  Vol100MB  datastore1  photon-6       Sat Sep 10 08:36:29 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  persistent             
N/A       vol1      datastore1  photon-6       Sat Sep 10 09:36:51 2016  detached                    N/A     500MB     23MB  thin         ext4             read-write  independent_persistent 
``` 

Step2: run `vmdkops_admin set` with non-exist tenant "DEFAULT0"
```
[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py set --volume=Vol100MB@datastore1 --tenant=_DEFAULT0 --options="access=read-only"
Failed to update access=read-only for Vol100MB@datastore1.
```

Step3: run `vmdkops_admin set` with exist tenant "_DEFAULT"
```
[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py set --volume=Vol100MB@datastore1 --tenant=_DEFAULT --options="access=read-only"
Successfully updated settings for : Vol100MB@datastore1
[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Tenant    Volume    Datastore   Created By VM  Created                   Attached To VM (name/uuid)  Policy  Capacity  Used  Disk Format  Filesystem Type  Access      Attach As              
--------  --------  ----------  -------------  ------------------------  --------------------------  ------  --------  ----  -----------  ---------------  ----------  ---------------------- 
_DEFAULT  vol1      datastore1  photon-6       Sat Sep 10 01:11:07 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  independent_persistent 
_DEFAULT  testvol   datastore1  photon-6       Sat Sep 10 04:21:02 2016  detached                    N/A     100MB     13MB  thin         ext4             read-write  independent_persistent 
_DEFAULT  Vol100MB  datastore1  photon-6       Sat Sep 10 08:36:29 2016  detached                    N/A     100MB     13MB  thin         ext4             read-only   persistent             
N/A       vol1      datastore1  photon-6       Sat Sep 10 09:36:51 2016  detached                    N/A     500MB     23MB  thin         ext4             read-write  independent_persistent 
 
[root@localhost:~]
```

File: esx_service/cli/vmdkops_admin.py [hunk 1]
```diff
@@ -527,14 +531,18 @@ def ls_dash_c(columns, tenant_reg):
 
 def all_ls_headers():
     """ Return a list of all header for ls -l """
-    return ['Volume', 'Datastore', 'Created By VM', 'Created',
+    return ['Tenant', 'Volume', 'Datastore', 'Created By VM', 'Created',
```
Comment (shuklanirdesh82) [inline esx_service/cli/vmdkops_admin.py]: I would vote for displaying Volume Name as the first column ... can you please accommodate the change?

File: esx_service/cli/vmdkops_admin.py [hunk 2]
```diff
@@ -415,6 +415,10 @@ def commands():
                     'help': 'Volume to set options for, specified as "volume@datastore".',
                     'required': True
                 },
+                '--tenant': {
+                    'help': 'Name of the tenant which volume to set belongs to.',
```
Comment (shaominchen) [inline esx_service/cli/vmdkops_admin.py]: Nit: Suggest to rephrase a little bit: "Name of the tenant the volume belongs to"

File: esx_service/utils/auth.py [hunk 3]
```diff
@@ -37,6 +37,7 @@
 DEFAULT_TENANT_UUID = '11111111-1111-1111-1111-111111111111'
 DEFAULT_DS  = '_DEFAULT'
 DEFAULT_DS_URL = DEFAULT_DS + "_URL"
+ORPHAN_TENANT = "_ORPHAN"
```
Comment (shaominchen) [inline esx_service/utils/auth.py]: Let's define this in auth_data_const.py instead.

File: esx_service/cli/vmdkops_admin.py (hunk 2 above)
Comment (lipingxue) [inline esx_service/cli/vmdkops_admin.py]: Done.

File: esx_service/utils/auth.py (hunk 3 above)
Comment (lipingxue) [inline esx_service/utils/auth.py]: Done.

**Comment by lipingxue** [comment]
After change the order/name of columns, the output looks like:
```
[root@localhost:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py ls
Volume    Datastore   VM-Group  Capacity  Used  Filesystem  Policy  Disk Format  Attached-to  Access      Attach-as               Created By  Created Date              
--------  ----------  --------  --------  ----  ----------  ------  -----------  -----------  ----------  ----------------------  ----------  ------------------------  
vol1      datastore1  _DEFAULT  100MB     13MB  ext4        N/A     thin         detached     read-write  independent_persistent  photon-6    Sun Sep 11 07:30:47 2016  
new-vol1  datastore1  N/A       100MB     13MB  ext4        N/A     thin         detached     read-write  independent_persistent  photon-6    Sun Sep 11 08:03:56 2016  
```

**Comment by msterin** [comment]
No more comments and looks good but - why the CI is failing ? Also , need conflict resolution.

File: esx_service/utils/auth.py (hunk 3 above)
Comment (lipingxue) [inline esx_service/utils/auth.py]: Done.

File: esx_service/cli/vmdkops_admin.py [hunk 4]
```diff
@@ -419,7 +419,7 @@ def commands():
                     'required': True
                 },
                 '--tenant': {
-                    'help': 'Name of the tenant which volume to set belongs to.',
+                    'help': 'Name of the tenant the volume belongs to.',
```
Comment (shaominchen) [inline esx_service/cli/vmdkops_admin.py]: Please make sure to resolve the conflicts with the other change which renames tenant to vm-group.

**Pull request merged**